apps/api/src/app/subscribers-v2/e2e/get-subscriber-preferences.e2e.ts

```diff
@@ -0,0 +1,133 @@
+import { randomBytes } from 'crypto';
+import { UserSession } from '@novu/testing';
+import { ChannelTypeEnum } from '@novu/shared';
+import { NotificationTemplateEntity } from '@novu/dal';
+import {
+  UpdateSubscriberGlobalPreferencesRequestDto,
+  UpdateSubscriberPreferenceRequestDto,
+  SubscriberResponseDto,
+} from '@novu/api/models/components';
+
+const v2Prefix = '/v2';
+let session: UserSession;
+
+describe('Get Subscriber Preferences - /subscribers/:subscriberId/preferences (GET) #novu-v2', () => {
+  let subscriber: SubscriberResponseDto;
+  let workflow: NotificationTemplateEntity;
+
+  beforeEach(async () => {
+    const uuid = randomBytes(4).toString('hex');
+    session = new UserSession();
+    await session.initialize();
+    subscriber = await createSubscriberAndValidate(uuid);
+    workflow = await session.createTemplate({
+      noFeedId: true,
+    });
+  });
+
+  it('should fetch subscriber preferences with default values', async () => {
+    const response = await session.testAgent.get(`${v2Prefix}/subscribers/${subscriber.subscriberId}/preferences`);
+
+    expect(response.statusCode).to.equal(200);
+    expect(response.body.data).to.have.property('global');
+    expect(response.body.data).to.have.property('workflows');
+
+    const { global, workflows } = response.body.data;
+
+    // Validate global preferences
+    expect(global).to.have.property('enabled');
+    expect(global).to.have.property('channels');
+    expect(global.enabled).to.be.true;
+
+    // Validate workflows array
+    expect(workflows).to.be.an('array');
+  });
+
+  it('should return 404 if subscriber does not exist', async () => {
+    const invalidSubscriberId = `non-existent-${randomBytes(2).toString('hex')}`;
+    const response = await session.testAgent.get(`${v2Prefix}/subscribers/${invalidSubscriberId}/preferences`);
+
+    expect(response.statusCode).to.equal(404);
+  });
+
+  it('should handle subscriber with modified workflow preferences', async () => {
+    // created workflow has 'email' and 'in-app'  channels enabled by default
+    const workflowId = workflow._id;
+
+    // disable email channel for this workflow
+    const enableEmailPreferenceData: UpdateSubscriberPreferenceRequestDto = {
+      channel: {
+        type: ChannelTypeEnum.EMAIL,
+        enabled: false,
+      },
+    };
+
+    // TODO: replace with v2 endpoint when available
+    await session.testAgent
+      .patch(`/v1/subscribers/${subscriber.subscriberId}/preferences/${workflowId}`)
+      .send({ ...enableEmailPreferenceData });
+
+    const response = await session.testAgent.get(`${v2Prefix}/subscribers/${subscriber.subscriberId}/preferences`);
+
+    const { global, workflows } = response.body.data;
+
+    expect(response.statusCode).to.equal(200);
+
+    expect(global.channels).to.deep.equal({ in_app: true, email: true });
+    expect(workflows).to.have.lengthOf(1);
+    expect(workflows[0].channels).to.deep.equal({ in_app: true, email: false });
```

Comment:
It seems like it's by default has `email: false` because in the next test, you don't update workflow preference, but the check is the same. So I would suggest changing maybe `in_app: false`

Reply:
I've changed these tests in PATCH PR because they were overlapping, instead I focused GET tests to behaviour unique to GET